x-pack/plugins/security/server/lib/parse_next.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { parse } from 'url';
+import { isInternalURL } from './is_internal_url';
+
+export function parseNext(href: string, basePath = '') {
+  const { query, hash } = parse(href, true);
+  if (!query.next) {
+    return `${basePath}/`;
+  }
+
+  let next: string;
```

Comment:
__note:__ we didn't have this check in 6.8 before, but it seems it's better to have it.